features/cellular/framework/AT/AT_CellularContext.cpp

```diff
@@ -659,17 +665,27 @@ nsapi_error_t AT_CellularContext::disconnect()
         // 3GPP TS 27.007:
         // For EPS, if an attempt is made to disconnect the last PDN connection, then the MT responds with ERROR
         if (_is_context_active && (rat < CellularNetwork::RAT_E_UTRAN || active_contexts_count > 1)) {
+            _at.clear_error();
             _at.cmd_start("AT+CGACT=0,");
             _at.write_int(_cid);
             _at.cmd_stop_read_resp();
         }
-    }
 
-    if (!_at.get_last_error()) {
-        _is_connected = false;
-        call_network_cb(NSAPI_STATUS_DISCONNECTED);
+        if (_new_context_set) {
+            _at.clear_error();
+            _at.cmd_start("AT+CGDCONT=");
+            _at.write_int(_cid);
+            _at.cmd_stop_read_resp();
+        }
+        _at.clear_error();
+        _at.cmd_start("AT+CGATT=0");
```

Comment:
I guess `CellularNetwork::detach` should be updated to detach from network.

Reply:
added register to CellularNetwork::detach